Ensure subp doesn't run during logs.py import

TheRealFalcon opened the pull request:
## Proposed Commit Message
<!-- Include a proposed commit message because PRs are squash merged
by default.

See https://www.conventionalcommits.org/en/v1.0.0/#specification
for our commit message convention.

If the change is related to a particular cloud or particular distro,
please include the "optional scope" in the summary line. E.g.,
feat(ec2): Add support for foo to the baz

Types used by this project:
feat, fix, docs, ci, test, refactor, chore
-->
```
See individual commits
```

## Additional Context
`tests/unittests/cmd/devel/test_logs.py` would fail if you ran the file on its own due to subp happening on import.

The 2nd commit is just some extra cleanup.

## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

holmanb commented on cloudinit/cmd/devel/logs.py:
```diff
@@ -27,8 +27,23 @@
+    instance_data_sensitive: str
+
+
+def get_log_paths(init: Optional[Init] = None) -> LogPaths:
```
+1 nice

holmanb commented on cloudinit/cmd/main.py:
```diff
@@ -1050,7 +1050,7 @@ def main(sysv_args=None):
             )
 
-            logs_parser(parser_collect_logs)
+            logs_parser(parser=parser_collect_logs)
```
was this intended to be in the cleanup commit?

holmanb commented on cloudinit/cmd/devel/logs.py:
```diff
@@ -293,7 +298,11 @@ def collect_logs(tarfile, include_userdata: bool, verbosity=0):
 def handle_collect_logs_args(name, args):
     """Handle calls to 'cloud-init collect-logs' as a subcommand."""
-    return collect_logs(args.tarfile, args.userdata, args.verbosity)
+    return collect_logs(
```
was this intended to be in the cleanup commit?

holmanb reviewed (commented):
@TheRealFalcon the PR looks good, but I think some of the changes meant for the fix PR ended up in the cleanup PR, and tests fail on the first commit:
```
FAILED tests/unittests/cmd/devel/test_logs.py::TestCollectInstallerLogs::test_include_installer_logs_when_present[no_files_include_userdata] - AttributeError: module 'cloudinit.cmd.devel.logs' has no attribute '_collect_installer_logs'. Did you mean: 'collect_installer_logs'?
FAILED tests/unittests/cmd/devel/test_logs.py::TestCollectInstallerLogs::test_include_installer_logs_when_present[no_files_exclude_userdata] - AttributeError: module 'cloudinit.cmd.devel.logs' has no attribute '_collect_installer_logs'. Did you mean: 'collect_installer_logs'?
FAILED tests/unittests/cmd/devel/test_logs.py::TestCollectInstallerLogs::test_include_installer_logs_when_present[files_and_dirs_include_userdata] - AttributeError: module 'cloudinit.cmd.devel.logs' has no attribute '_collect_installer_logs'. Did you mean: 'collect_installer_logs'?
FAILED tests/unittests/cmd/devel/test_logs.py::TestCollectInstallerLogs::test_include_installer_logs_when_present[files_and_dirs_exclude_userdata] - AttributeError: module 'cloudinit.cmd.devel.logs' has no attribute '_collect_installer_logs'. Did you mean: 'collect_installer_logs'?
```

TheRealFalcon commented:
I had to rebase because local linting was acting weird, so the most recent changes might not be obvious. I'll explain them here.

For your two comments of `was this intended to be in the cleanup commit?`, the answer is no actually. The first commit had initially also moved arg parsing around, but I later noticed it broke some other code so I undid most of it, but the named arguments stuck around. But I agree that they make more sense in the other commit, so I moved them.

Beyond those two, the other change was the test_logs.py change of moving the `logs.collect_installer_logs` -> `logs._collect_installer_logs` to the later commit, allowing unit tests to pass cleanly on both commits.

I also amended the commit message of both commits to include the PR number here.

holmanb reviewed (approved):
Nit: I would use `fix` rather than `refactor` for the first commit, since it makes functional changes which resolve undesirable behavior. Otherwise this looks good to me!
